public/app/plugins/panel/debug/panelcfg.cue

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Grafana Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grafanaplugin
+
+import (
+	// "github.com/grafana/grafana/packages/grafana-schema/src/common"
+	"github.com/grafana/grafana/pkg/plugins/pfs"
+)
+
+// This file (with its sibling .cue files) implements pfs.GrafanaPlugin
+pfs.GrafanaPlugin
+
+composableKinds: PanelCfg: {
+	maturity: "experimental"
+
+	lineage: {
+		seqs: [
+			{
+				schemas: [
+					{
+						UpdateConfig: {for k, _ in UpdateCounters {"\(k)": bool}} @cuetsy(kind="type")
+
+						UpdateCounters: {
```

Comment:
I think this is good -- but slightly weird and perhaps worth pointing out that `UpdateCounters` is not part of saved schema contract (it is only used at runtime, and not within the options)

Reply:
Hmm... any better if we only define `UpdateConfig` in this cue file then import it in a TS file and do the following?

```ts
export type UpdateCounters = {
  [K in keyof UpdateConfig]: number;
};
```